domain/annotation/state/query.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"fmt"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/annotations"
+	annotationerrors "github.com/juju/juju/domain/annotation/errors"
+)
+
+// getAnnotationQueryForID provides a query for the given id, based on pre-computed queries for
+// GetAnnotations for different kinds of ids. We keep these static (avoiding dynamically generating
+// table names and fields) to keep things safe.
+func getAnnotationQueryForID(id annotations.ID) string {
+	var query string
+	switch id.Kind {
+	case annotations.KindModel:
+		query = `SELECT (key, value) AS (&Annotation.*) from annotation_model`
+	case annotations.KindMachine:
+		query = `
+		SELECT (key, value) AS (&Annotation.*)
+		FROM annotation_machine
+		WHERE machine_uuid = $M.uuid
+		`
+	case annotations.KindUnit:
+		query = `
+		SELECT (key, value) AS (&Annotation.*)
+		FROM annotation_unit
+		WHERE unit_uuid = $M.uuid
+		`
+	case annotations.KindApplication:
+		query = `
+		SELECT (key, value) AS (&Annotation.*)
+		FROM annotation_application
+		WHERE application_uuid = $M.uuid
+		`
+	case annotations.KindStorage:
+		query = `
+		SELECT (key, value) AS (&Annotation.*)
+		FROM annotation_storage_instance
+		WHERE storage_instance_uuid = $M.uuid
+		`
+	case annotations.KindCharm:
+		query = `
+		SELECT (key, value) AS (&Annotation.*)
+		FROM annotation_charm
+		WHERE charm_uuid = $M.uuid
+		`
+	}
+	return query
+}
+
+// setAnnotationQueryForID provides a query for the given id, based on pre-computed queries for
+// SetAnnotations for different kinds of ids. We keep these static (avoiding dynamically generating
+// table names and fields) to keep things safe.
+func setAnnotationQueryForID(id annotations.ID) string {
+	var query string
+	switch id.Kind {
+	case annotations.KindModel:
+		query = `INSERT INTO annotation_model (key, value)
+		VALUES ($M.key, $M.value)
+		  ON CONFLICT(key) DO UPDATE SET value=$M.value`
```

Comment:
Same goes for these queries.

```suggestion
		query = `
INSERT INTO annotation_model (key, value)
VALUES ($M.key, $M.value)
ON CONFLICT(key) DO UPDATE SET value=$M.value`
```